src/network/wait-online/manager.c

```diff
@@ -91,25 +91,22 @@ static int manager_link_is_online(Manager *m, Link *l, LinkOperationalStateRange
                                             "link is being processed by networkd: setup state is %s.",
                                             l->state);
 
-        if (s.min < 0)
-                s.min = m->required_operstate.min >= 0 ? m->required_operstate.min
-                                                       : l->required_operstate.min;
+        if (!s || !operational_state_range_is_valid(s))
```

Comment:
if you make operational_state_range_is_valid() return false in case of NULL, you could simplify this by dropping the left-hand side